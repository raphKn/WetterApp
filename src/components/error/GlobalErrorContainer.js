import { connect } from 'react-redux';
import { createResetGlobalError, getGlobalErrorMessage } from '../../modules/globalError';
import GlobalError from './GlobalError';

const mapStateToProps = state => ({
    message: getGlobalErrorMessage(state),
});

const mapDispatchToProps = dispatch => ({
    cancelError: () => {
        dispatch(createResetGlobalError());
    },
});

const GlobalErrorContainer = connect(mapStateToProps, mapDispatchToProps)(GlobalError);

export default GlobalErrorContainer;