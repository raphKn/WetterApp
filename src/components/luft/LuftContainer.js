import { connect } from 'react-redux';
import Luft from './Luft';
import { getLuft } from '../../modules/luft';

const mapStateToProps = state => ({
    luft: getLuft(state),
});

const LuftContainer = connect(mapStateToProps, null)(Luft);

export default LuftContainer;