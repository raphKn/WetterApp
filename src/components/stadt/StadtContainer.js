import { connect } from 'react-redux';
import { getStadt } from '../../modules/stadt';
import Stadt from './Stadt';

const mapStateToProps = state => ({
    stadt: getStadt(state),
});

const ComponentContainer = connect(mapStateToProps, null)(Stadt);

export default ComponentContainer;