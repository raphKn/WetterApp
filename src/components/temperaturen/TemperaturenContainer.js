import { connect } from 'react-redux';
import { getTemperaturen } from '../../modules/temperaturen';
import Wetter from './Temperaturen';

const mapStateToProps = state => ({
    temperaturen: getTemperaturen(state),
});

const TemperaturenContainer = connect(mapStateToProps, null)(Wetter);

export default TemperaturenContainer;
