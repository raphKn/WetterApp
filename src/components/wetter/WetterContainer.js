import { connect } from 'react-redux';
import { getWetter } from '../../modules/wetter';
import Wetter from './Wetter';

const mapStateToProps = state => ({
    wetter: getWetter(state),
});

const WetterContainer = connect(mapStateToProps, null)(Wetter);

export default WetterContainer;