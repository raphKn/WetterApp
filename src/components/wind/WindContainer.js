import { connect } from 'react-redux';
import { getWind } from '../../modules/wind';
import Wind from './Wind';

const mapStateToProps = state => ({
    wind: getWind(state),
});

const ComponentContainer = connect(mapStateToProps)(Wind);

export default ComponentContainer;
