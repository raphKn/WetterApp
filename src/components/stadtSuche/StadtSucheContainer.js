import { connect } from 'react-redux';
import { getStadtSuche } from '../../modules/stadtSuche';
import { createWetterSuche } from '../../sagas/wetter/wetterSaga';
import StadtSuche from './StadtSuche';

const mapStateToProps = state => ({
    defaultText: getStadtSuche(state),
});

const mapDispatchToProps = dispatch => ({
    onSubmit: stadt => dispatch(createWetterSuche(stadt)),
});

const StadtSucheContainer = connect(mapStateToProps, mapDispatchToProps)(StadtSuche);

export default StadtSucheContainer;
