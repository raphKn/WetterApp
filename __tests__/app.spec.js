import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from '../src/app';
import StadtSucheContainer from '../src/components/stadtSuche/StadtSucheContainer';
import StadtContainer from '../src/components/stadt/StadtContainer';
import WindContainer from '../src/components/wind/WindContainer';
import TemperaturenContainer from '../src/components/temperaturen/TemperaturenContainer';
import LuftContainer from '../src/components/luft/LuftContainer';
import WetterContainer from '../src/components/wetter/WetterContainer';
import GlobalErrorContainer from '../src/components/error/GlobalErrorContainer';

jest.mock('../src/components/stadtSuche/StadtSucheContainer');
jest.mock('../src/components/stadt/StadtContainer');
jest.mock('../src/components/wind/WindContainer');
jest.mock('../src/components/temperaturen/TemperaturenContainer');
jest.mock('../src/components/luft/LuftContainer');
jest.mock('../src/components/wetter/WetterContainer');
jest.mock('../src/components/error/GlobalErrorContainer');

describe('Component', () => {
    it('renders in a snapshot', () => {
        const wrapper = renderer.create(
            <App />).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
    it('renders all subinfos', () => {
        const wrapper = shallow(
            <App />);
        expect(wrapper.find(StadtSucheContainer).length).toBe(1);
        expect(wrapper.find(StadtContainer).length).toBe(1);
        expect(wrapper.find(WindContainer).length).toBe(1);
        expect(wrapper.find(TemperaturenContainer).length).toBe(1);
        expect(wrapper.find(LuftContainer).length).toBe(1);
        expect(wrapper.find(WetterContainer).length).toBe(1);
        expect(wrapper.find(GlobalErrorContainer).length).toBe(1);
    });
});
