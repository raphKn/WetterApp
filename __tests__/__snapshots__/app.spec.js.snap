// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders in a snapshot 1`] = `
<RCTScrollView>
  <View>
    <Image
      resizeMode="contain"
      source={1}
      style={
        Object {
          "height": 375,
          "width": 750,
        }
      }
    />
  </View>
</RCTScrollView>
`;
